roadmap/implementors-guide/subsystems-overview.md

```diff
@@ -0,0 +1,95 @@
+# Subsystems Overview
+
+These subsystems comprise the Validity module. All detailed documentation lives within the guide; this document simply attempts to enumerate and briefly describe each subsystem and the map of their communications.
+
+- Overseer
+- Candidate Selection
+- Candidate Validation
+- Candidate Backing
+- Statement Distribution
+- Misbehavior Arbitration
+
+## Overseer
+
+The overseer is the postal service: it relays messages between the other subsystems. At any time in this document when we state that a subsystem sends a message to another, it does so via the overseer. The overseer also alerts other subsystems when they should be prepared to work on new relay parents, and when they should shut down preparations to work on old, finalized, or otherwise obsolete relay parents.
+
+## Candidate Selection
+
+The candidate selection subsystem is in charge of seconding 0 or 1 parablock candidates per relay parent.
+
+### Inbound Messages
+
+- **Overseer**: a new parablock candidate has arrived from a collator.
+
+### Outbound Messages
+
+- **Candidate Backing**: this candidate (by hash) satisfies system properties; please validate it
+
+## Candidate Backing
+
+The candidate backing subsystem is in charge of producing statements which assert the (non-)validity of a parachain candidate.
+
+### Inbound Messages
+
+- **Candidate Selection**: a candidate which satisfies system properties
+- **Candidate Validation**: a particular candidate is valid, or not
+
+### Outbound Messages
+
+- **Candidate Validation**: is this candidate in fact valid?
+- **Statement Distribution**: I testify that this candidate is (in-)valid
+
+## Candidate Validation
+
+The candidate validation subsystem handles the details of validating a candidate. For any given parachain candidate, it arranges the appropriate context: the validation function, the relay parent head data, the in-progress relay data, etc.
+
+### Inbound Messages
+
+- **Various Subsystems**: requests to validate a particular parablock candidate
```

Comment:
[Previous comment](https://github.com/w3f/research/pull/87#pullrequestreview-415738247):

> @rphmeier 
> one open question is how the PoV fetch is handled.
> 
> in the case of validating a block from a collator (to decide whether to second), the collator has sent us a PoV, so we could send it directly to the validation subsystem. In the case of validating a candidate seconded by another validator, we just have the receipt and not the PoV, so we'd want to proceed through these steps:
>
> 1. Figure out if we've already validated the candidate
> 1. If not, fetch PoV
> 1. Upon PoV fetch, do validation
> 
> We might want something like a PoVOrigin sent alongside a validation request that tells us where to get the PoV. Included would be one option, and fetching from the network would be another. But maybe this is where things get complicated, since the network might want to use information about statements received from other validators to figure out who might have the PoV. At the moment, we'll just gossip all PoVs in the same subsystem as statements, so we can defer this to later.

Reply:
ed5f4696